### PR TITLE
Feature/i7 extensions

### LINKED ISSUE #6: Improve run command execution.
Current run functionality is not processing files correctly. 

One way to fix this could be custom tasks that would handle this better.

### LINKED ISSUE #7: Include extensions in code-quality package.
It is hard to maintain and update package if every extension is separate package. They should all be included in one package.

### PR DESCRIPTION
**Pull request by guncha25**
**Custom tasks.**

**Features:**
- Adds all tasks from composer directly in src.
- Refactor tasks to use correct files per context.
- Adds phpcs override. 

**Expected behaviour:**
- Running in git context only files with correct extension, staged, in defined path (run_on) and not containing ignore pattern should be checked.
- Running in run context all files with correct extension, in defined path (run_on) and not containing ignore pattern should be checked.

**Test:**
```
composer require wunderio/code-quality:dev-feature/i7-extensions --dev
cp vendor/wunderio/code-quality/grumphp.yml.dist ./grumphp.yml
```
- Try default configurations. (No extra arguments to task in grumphp.yml)

Todo:
- [x] Rewrite phpcs task to support files in run context.

Fixes #7 
Fixes #6

### CODE REVIEW

File: src/Task/Ecs/EcsTask.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfigurableOptions(): OptionsResolver
+  {
+    $resolver = new OptionsResolver();
+    $resolver->setDefaults([
+      'ignore_patterns' => ['*/vendor/*','*/node_modules/*', '*/core/*', '*/modules/contrib/*', '*/themes/contrib/*'],
```
Comment (mgalang) [inline src/Task/Ecs/EcsTask.php]: I wonder if these ignore patterns and other parameter values could be just declared as a class static attribute?

Comment (guncha25) [reply src/Task/Ecs/EcsTask.php]: Good point. Probably in base class. Repeating patterns, extensions should definitely be added in single place.

File: src/Task/CheckFilePermissions/CheckFilePermissionsTask.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  {
+    $config = $this->getConfiguration();
+    $files = $this->getFiles($context, $config);
+
+    if (0 === \count($files)) {
+      return TaskResult::createSkipped($this, $context);
+    }
+
+    $arguments = $this->processBuilder->createArgumentsForCommand('check_perms');
+    if (!$context instanceof GitPreCommitContext) {
```
Comment (mgalang) [inline src/Task/CheckFilePermissions/CheckFilePermissionsTask.php]: What if this whole "if" block is refactored to its own `getFiles` method?

So it just becomes:

```
$files = $this->getFiles();
$arguments->addFiles($files);
```

File: bin/check_perms [hunk 3]
```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+
+set -e
+set -u
+export PATH=/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin
```
Comment (mgalang) [inline bin/check_perms:5]: Is this actually executed on local? This somehow overrides my path envs in the current terminal session.

Comment (guncha25) [reply bin/check_perms:5]: @ragnarkurmwunder ^

Comment (guncha25) [reply bin/check_perms:5]: I don't really know how this works. Its something that probably could be separately fixed. It's just a copy from grumphp-file-permissions package.

Comment (mgalang) [reply bin/check_perms:5]: At least It doesn't seem to be overriding, I tested in my local

**Review by mgalang — APPROVED** [review]
Tested in local. Didn't have any issues.
Only found a few minor issues in code.
Approving this for my part.

File: src/Task/CheckFilePermissions/CheckFilePermissionsTask.php (hunk 2 above)
Comment (guncha25) [reply src/Task/CheckFilePermissions/CheckFilePermissionsTask.php]: Refactored everything :) Now you need to test again.

**Review by hkirsman — COMMENTED** [review]
This is the lost review :(

**Pull request merged**